react-web/src/components/ConnectWallet/ConnectWallet.tsx

```diff
@@ -36,13 +36,14 @@ const ConnectWallet = () => {
         }
     }
 
-    useEffect(() => {
+    const bindAccountChange = () => {
         if (CardanoNS?.onAccountChange && typeof CardanoNS.onAccountChange === 'function') { 
             CardanoNS.onAccountChange((address: Array<any>) => {
                 setAddress(address[0]);
             })
         }
-    });
+    }
+    bindAccountChange();
```

Comment:
this is still equivalent to what we had before: For every re-render of ConnectWallet, the bindAccountChange() function will repeatedly be called. Is this what we want to happen or will there be enough to call that just when the CardanoNS reference changes? 
I'm afraid that by subscribing to `onAccountChange` multiple times, `setAddress(address[0])`, when the account changes, will be called as much as the total number of previous re-renders we had before.

Reply:
Replaced listening to onAccountChange with a manual check to see if the address has changed. That is again with a use of [] useEffect; but seems like the quickest way to do so.